tests/Integration/CompatibilityTest.php

```diff
@@ -24,17 +24,21 @@ public function test_should_plugin_load(): void {
 		$is_gutenberg_plugin_active = self::get_method( 'is_gutenberg_plugin_active', Compatibility::class );
 		$should_plugin_load = self::get_method( 'should_plugin_load', Compatibility::class );
 
-		self::assertTrue( $is_websocket_url_defined->invoke( null ), 'is_websocket_url_defined() should be true' );
-
 		// Start with Gutenberg deactivated.
 		deactivate_plugins( 'gutenberg/gutenberg.php' );
 
 		self::assertFalse( $is_gutenberg_plugin_active->invoke( null ), 'is_gutenberg_plugin_active should be false' );
+		// Regardless of the WebSocket URL value, the plugin should not load when Gutenberg is not active.
 		self::assertFalse( $should_plugin_load->invoke( null ), 'should_plugin_load() should be false' );
 
 		// Activate Gutenberg.
 		activate_plugin( 'gutenberg/gutenberg.php' );
 		self::assertTrue( $is_gutenberg_plugin_active->invoke( null ), 'is_gutenberg_plugin_active should be true' );
-		self::assertTrue( $should_plugin_load->invoke( null ), 'should_plugin_load() should be true' );
+
+		if ( true === $is_websocket_url_defined->invoke( null ) ) {
+			self::assertTrue( $should_plugin_load->invoke( null ), 'should_plugin_load() should be true' );
+		} else {
+			self::assertFalse( $should_plugin_load->invoke( null ), 'should_plugin_load() should be false' );
+		}
```

Comment:
O_o

We should either require that the constant be defined or mock the constant definition and test both cases.

Reply:
I'm not clear on what is being proposed here. The constant can be `undefined`,  `''` or a non-empty string value. In the first two cases `is_websocket_url_defined()` returns `false`.